WebService/v1/Filters/AuthorizeActionFilterAttribute.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Azure.IoTSolutions.UIConfig.Services.Exceptions;
+using Microsoft.Azure.IoTSolutions.UIConfig.WebService.Auth;
+
+namespace Microsoft.Azure.IoTSolutions.UIConfig.WebService.v1.Filters
+{
+    public class AuthorizeAttribute : TypeFilterAttribute
+    {
+        public AuthorizeAttribute(string allowedActions)
+            : base(typeof(AuthorizeActionFilterAttribute))
+        {
+            this.Arguments = new object[] { allowedActions };
+        }
+    }
+
+    public class AuthorizeActionFilterAttribute : Attribute, IAsyncActionFilter
+    {
+        private readonly string allowedAction;
+
+        public AuthorizeActionFilterAttribute(string allowedAction)
+        {
+            this.allowedAction = allowedAction;
+        }
+
+        public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        {
+
+            bool isAuthorized = this.ValidateAuthorization(context.HttpContext, this.allowedAction);
+
+            if (!isAuthorized)
+            {
+                throw new NoAuthorizationException($"Current user is not authorized to perform this action: '{this.allowedAction}'");
+            }
+            else
+            {
+                await next();
+            }
+        }
+
+        /// <summary>
+        /// Validate allowed actions of current user based on role claims against the declared actions
+        /// in the Authorize attribute of controller. The allowed action is case insensitive.
+        /// </summary>
+        /// <param name="httpContext">current context of http request</param>
+        /// <param name="allowedAction">allowed action required by controller</param>
+        /// <returns>true if validatation succeed</returns>
+        private bool ValidateAuthorization(HttpContext httpContext, string allowedAction)
```

Comment:
The naming semantics are off here.
If it returns a boolean, the method name should ask a true/false question: IsAuthorized.
A method named ValidateAuthorization sounds like it will throw an exception.  This does not.

Reply:
This is the naming from iot hub manager. I agree IsValidAuthorization would make more sense, but it might be better to stay consistent

Reply:
We should rename this to IsValidAuthorization once we are new repo.
Can you make a list of naming changes that we need to make in a single bug, so far i count 2?

Reply:
I think ValidateAuthorization() return true/false make sense to me just as ValidateToken() method. I don't think it is a bug.